NitroxServer/Serialization/World/WorldPersistence.cs

```diff
@@ -93,26 +93,38 @@ internal Optional<World> LoadFromFile(string saveDir)
                 return Optional.Empty;
             }
 
-            try
+            UpgradeSave(saveDir);
```

Comment:
The UpgradeSave can throw null ref when `saveFileVersion` is null. This should probably be handled to run a `return` early.